internal/pkg/cluster/check/service.go

```diff
@@ -22,14 +22,13 @@ func ServiceStateAssertion(ctx context.Context, cluster ClusterInfo, service str
 		return err
 	}
 
-	// perform check against "init" node
-	initNodes := cluster.NodesByType(runtime.MachineTypeInit)
+	nodes := cluster.NodesByType(runtime.MachineTypeControlPlane)
 
-	if len(initNodes) != 1 {
-		return fmt.Errorf("init node not found, len(initNodes) = %d", len(initNodes))
+	if len(nodes) == 0 {
+		return fmt.Errorf("no control plane nodes found")
 	}
 
-	nodeCtx := client.WithNodes(ctx, initNodes[0])
+	nodeCtx := client.WithNodes(ctx, nodes[0])
```

Comment:
we need "bootstrap" node here iirc, as we're checking for `bootkube`, and bootkube only runs on bootstrap/init node. should we also enforce some way we pick up bootstrap node? 

and as init node is technically still supported, we could do the following:

1. if init node is defined, use it
2. otherwise, pick the first control plane node with the first IP in alphabetical order (and make sure we call bootstrap on that very same node).

Reply:
Yeah I wasn't quite sure how to handle this. The problem I seem to be having now is that `nodes[0] could be any one of the control plane nodes. Alphabetical order should fix this.

Reply:
I made these changes, and that gets through the initial bootstrapping but running cluster health checks after reboot/reset tests complains about not being able to find the bootkube service. Oh, now that I am saying it out loud, the bootkube service isn't created ever again. That is the whole point of the bootstrap API. Any ideas on what we should do here? I am thinking that maybe the bootkube service check is optional in some way.